source/1_engine/rex_engine/include/rex_engine/diagnostics/verify.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "rex_engine/diagnostics/assert.h"
+
+namespace rex
+{
+  void rex_verify(const rsl::fmt_stack_string& msg);
+} // namespace rex
+
+#define REX_VERIFY(...)                                                                                                                                                                                                                                  \
+  rex::rex_verify(rsl::format(__VA_ARGS__));                                                                                                                                                                                                             \
+  while(true)                                                                                                                                                                                                                                            \
+  {                                                                                                                                                                                                                                                      \
+    DEBUG_BREAK();                                                                                                                                                                                                                                       \
```

Comment:
this looks to be a copy of the assert implementation with just an extra thing to open a dialog
can't it use the existing assert implementation and show the messagebox afterwards

Reply:
the library seems pretty small so maybe we'll port it into the engine

Reply:
I could yes but that looked a bit confusing to me to be honest as we created verify to have a check in release cuz asserts are compiled out but then we still call assert code in release. I can reuse the code but it felt contradicting. 

I don't wanna fall into the spdlog trap again with this.
By reinventing the wheel again and implementing this ourselves.

Reply:
yes, it shouldn't use be called "rex_assert" then, you can rename the func to something more generic and use that instead. can put a comment there that it's used by both assert and verify